stl/inc/xlocale

```diff
@@ -1989,33 +1988,123 @@ protected:
         _Adl_verify_range(_First2, _Last2);
         _Mid1 = _First1;
         _Mid2 = _First2;
-        for (;;) {
-            if (_Mid1 == _Last1) {
-                return ok;
+
+        // Handle 0 length string
+        if (_Mid1 == _Last1)
+            return ok;
+
+        // "C" -> Unicode direct translation
+        if (_Cvt._Isclocale) {
+            while (true) {
+                if (_Mid2 == _Last2)
+                    return partial;
+
+                *_Mid2++ = static_cast<unsigned char>(*_Mid1++);
+
+                if (!(*_Mid2) || _Mid1 == _Last1)
+                    return ok;
             }
+        }
 
-            if (_Mid2 == _Last2) {
-                return partial;
+        // Else
+        if (_Cvt._Mbcurmax == 4) {

```

Comment:
let's not duplicate this code.

Additioanlly I don't think we can make this change at all. It's a massive, massive change in behavior

Reply:
actually at some point we should try and reduce the number of utf-8 conversion functions sitting around, we already have at least two (there's one in format)